File: C#/RemoveNthFromEnd.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode RemoveNthFromEnd(ListNode head, int n) {
        
        List<int> Values = new List<int>();
        ListNode Node = head;
        int sz = 0;
        
        while (Node != null)
        {
            Values.Add(Node.val);
            
            sz++;
            Node = Node.next;
        }
        
        // Default
        if (sz == 1)
        {
            return null;
        }
        
        Values.RemoveAt(sz - n);
        
        ListNode PreviousNode = new ListNode(Values[Values.Count - 1], null);
        ListNode Output = PreviousNode;
        for (int i = Values.Count - 2; i > -1; i--)
        {
            Output = new ListNode(Values[i], PreviousNode);;
            PreviousNode = Output;
        }
        
        return Output;        
    }
}

File: Python/removeNthFromEnd.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:

        valList = []

        while head:
            valList.append(head.val)
            head = head.next

        valList.pop(len(valList) - n)

        valList.reverse()
        curr, prev = None, None

        for val in valList:
            curr = ListNode(val, prev)
            prev = curr

        return curr